packages/flutter/lib/src/services/autofill.dart

```diff
@@ -0,0 +1,287 @@
+  /// string value is not recommended as it may not be understood by the platform.
+  /// Each hint in the list, if not ignored, will be translated to the platform's
+  /// autofill hint type understood by its autofill services:
+  ///
+  /// * On iOS, only the first hint in the list is accounted for. The hint will
+  /// be translated to a [UITextContentType](https://developer.apple.com/documentation/uikit/uitextcontenttype).
+  ///
+  /// * On Android, all hints in the list are translated to Android hint strings.
+  ///
+  /// * On web, only the first hint is accounted for and will be translated to
+  /// an "autocomplete" string.
+  ///
+  /// See also:
+  ///
+  /// * [AutofillHints], a list of autofill hint strings that is predefined on at
+  /// least one platform.
+  ///
+  /// * [UITextContentType](https://developer.apple.com/documentation/uikit/uitextcontenttype),
+  /// the iOS equivalent.
+  ///
+  /// * Android [autofillHints](https://developer.android.com/reference/android/view/View#setAutofillHints(java.lang.String...)),
+  /// the Android equivalent.
+  ///
+  /// * The [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) attribute,
+  /// the web equivalent.
+  /// {@endtemplate}
+  final List<String> autofillHints;
+
+  /// The current [TextEditingValue] of the [AutofillClient].
+  final TextEditingValue currentEditingValue;
+
+  /// Returns a representation of this object as a JSON object.
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'uniqueIdentifier': uniqueIdentifier,
+      'hints': autofillHints,
+      'editingValue': currentEditingValue.toJSON(),
+    };
+  }
+}
+
+/// An object that represents an autofillable input field in the autofill workflow.
+///
+/// An [AutofillClient] provides autofill-related information of the input field
+/// it represents to the platform, and consumes autofill inputs from the platform.
+abstract class AutofillClient {
+  /// The unique identifier of this [AutofillClient].
+  ///
+  /// Must not be null;
+  String get autofillId;
+
+  /// The [TextInputConfiguration] that describes this [AutofillClient].
+  TextInputConfiguration get textInputConfiguration;
+
+  /// Requests this [AutofillClient] update its [TextEditingState] to the given
+  /// state.
+  void updateEditingValue(TextEditingValue newEditingValue);
+}
+
+/// A [TextInputClient] that triggers autofill when focused.
+///
+/// This is typically a [State] or [Element], and should be able to gain focus in
+/// order to trigger autofill.
+abstract class AutofillTrigger extends TextInputClient {
+  /// The [AutofillScope] that this [AutofillTrigger] belongs to.
+  ///
+  /// This [AutofillTrigger] will participate in autofill alone if null.
+  AutofillScope get currentAutofillScope;
+}
+
+// Generic mixin for [AutofillClient]s.
+//
+// Provides a default implementation for [AutofillClient.autofillId] that extracts
```

Comment:
For documenting mixins use doc comments instead: https://dart.dev/guides/language/language-tour#documentation-comments